src/Nethermind/Nethermind.Trie/NibbleExtensions.cs

```diff
@@ -155,25 +156,148 @@ public static byte[] ToCompactHexEncoding(byte[] nibbles)
 
         public static byte[] NibblesToByteStorage(Span<byte> nibbles)
         {
-            int oddity = nibbles.Length % 2;
-            byte[] bytes = new byte[nibbles.Length / 2 + 1];
-            for (int i = 0; i < bytes.Length - 1; i++)
+            //int oddity = nibbles.Length % 2;
+            //byte[] bytes = new byte[nibbles.Length / 2 + 1];
+            //for (int i = 0; i < bytes.Length - 1; i++)
+            //{
+            //    bytes[i + 1] = ToByte(nibbles[2 * i + oddity], nibbles[2 * i + 1 + oddity]);
+            //}
+            //if (oddity == 1)
+            //{
+            //    bytes[0] = ToByte(1, nibbles[0]);
+            //}
+            //return bytes;
+
+            //int oddity = nibbles.Length % 2;
+            //byte[] bytes = new byte[nibbles.Length / 2 + 1];
+            //for (int i = 0; i < bytes.Length - 1; i++)
+            //{
+            //    bytes[i] = ToByte(nibbles[2 * i], nibbles[2 * i + 1]);
+            //}
+            //if (oddity == 1)
+            //{
+            //    bytes[^1] = ToByte(nibbles[^1], 0);
+            //}
+            //else
+            //{
+            //    bytes[^1] = 255;
+            //}
+            //return bytes;
+
+
+            //return nibbles.ToArray();
+
+            Span<byte> bytes = stackalloc byte[nibbles.Length];
```

Comment:
I think it might be better to just allocate the max value by const - it allows .net to optimize it better when it is a const.

Also add SkipLocalsInit to not 0 the stack.